mvpa2/datasets/gifti.py

```diff
@@ -0,0 +1,190 @@
+                             'NIFTI_INTENT_NODE_INDEX',
+                             'NIFTI_INTENT_SHAPE')
+
+    return intent_string not in not_data_intent_strings
+
+def _gifti_intent_is_node_indices(intent_string):
+    return intent_string == 'NIFTI_INTENT_NODE_INDEX'
+
+
+def _get_gifti_image(samples):
+    if isinstance(samples, basestring):
+        samples = giftiio.read(samples)
+
+    required_class=gifti.GiftiImage
+    if not isinstance(samples, required_class):
+        raise TypeError('Input of type %s must be a %s' %
+                                    (samples, required_class))
+
+    return samples
+
+def gifti_dataset(samples, targets=None, chunks=None):
+    """
+    Parameters
+    ----------
+    samples : str or GiftiImage
+      GIFTI surface-based data, specified either as a filename or an image.
+    targets : scalar or sequence
+      Label attribute for each volume in the timeseries.
+    chunks : scalar or sequence
+      Chunk attribute for each volume in the timeseries.
+    """
+    node_indices = None
+    data_vectors = []
+    intents = []
+
+    image = _get_gifti_image(samples)
+
+    for darray in image.darrays:
+        intent_string=_gifti_intent_niistring(darray.intent)
+
+        if _gifti_intent_is_data(intent_string):
+            data_vectors.append(darray.data)
+            intents.append(intent_string)
+
+        elif _gifti_intent_is_node_indices(intent_string):
+            node_indices=darray.data
+
+
+
+    samples = np.asarray(data_vectors)
+    nsamples, nfeatures = samples.shape
+
+    # set sample attributes
+    sa = SampleAttributesCollection(length=nsamples)
+
+    sa['intents'] = intents
+
+    if targets is not None:
+        sa['targets'] = targets
+
+    if chunks is not None:
+        sa['chunks'] = chunks
+
+    # set feature attributes
+    fa = FeatureAttributesCollection(length=nfeatures)
+
+    if node_indices is not None:
+        fa['node_indices'] = node_indices
+
+    return Dataset(samples=samples, sa=sa, fa=fa)
+
+
+def map2gifti(ds, filename=None, encoding='GIFTI_ENCODING_B64GZ'):
+    """Maps data(sets) into a GiftiImage, and optionally saves it to disc.
+
+    Parameters
+    ----------
+    ds : Dataset or numpy.ndarray
+      The data to be mapepd
+    filename : basestring or None, optional
+      Filename to which the GiftiImage is stored
+    encoding : "ASCII" or "Base64Binary" or "GZipBase64Binary", optional
+      Encoding format of data
+
+    Returns
+    -------
+    img : GiftiImage
+      dataset contents represented in GiftiImage
+    """
+
+    darrays=[]
+
+    if isinstance(ds, np.ndarray):
+        samples=ds
+    elif isinstance(ds, Dataset):
+        samples=ds.samples
+    else:
+        raise TypeError('first argument must be Dataset or numpy.ndarray')
+
+    [nsamples,nfeatures]=samples.shape
+
+    def _get_attribute_value(ds, attr_name, keys_):
+        try:
+            attr_collection=ds.__dict__.get(attr_name)
+        except:
```

Comment:
isn't it PEP8 or whatever to have at least "exception Exception"? but here it should be a `KeyError`?the point is that it might mask some other failure, not just that the key is not there